transform-cli/src/main/java/com/ubercab/transform/CliTransform.java

```diff
@@ -0,0 +1,156 @@
+package com.ubercab.transform;
+
+import com.android.annotations.NonNull;
+import com.android.build.api.transform.Transform;
+import com.ubercab.transform.loader.SystemClassLoader;
+import com.ubercab.transform.runner.TransformRunner;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+/**
+ * Main entry point for the cli application to apply the transform.
+ */
+public class CliTransform {
+
+    @NonNull private static final String ENV_VAR_IN_JARS_DIR = "IN_JARS_DIR";
+    @NonNull private static final String ENV_VAR_OUT_JARS_DIR = "OUT_JARS_DIR";
+    @NonNull private static final String ENV_VAR_CONFIG_FILE = "CONFIG_FILE";
+    @NonNull private static final String ENV_VAR_ANDROID_BOOTCLASSPATH = "ANDROID_BOOTCLASSPATH";
+    @NonNull private static final String ENV_VAR_TRANSFORM_DEPENDENCIES_FILE = "TRANSFORM_JARS_FILE";
+    @NonNull private static final String ENV_VAR_TRANSFORM_CLASS = "TRANSFORM_CLASS";
+    @NonNull private static final String DEPENDENCIES_SEPARATOR = ":";
+
+    private CliTransform() {
+    }
+
+    /**
+     * Main.
+     *
+     * @param args arguments.
+     */
+    public static void main(@NonNull String[] args) {
+        if (args.length != 0) {
+            StringBuilder sb = new StringBuilder();
+            sb.append("No argument is expected. All parameters should be passed through env vars.\n");
+            sb.append(ENV_VAR_IN_JARS_DIR + " : jars input directory\n");
+            sb.append(ENV_VAR_OUT_JARS_DIR + " : jars output directory\n");
+            sb.append(ENV_VAR_CONFIG_FILE + " : configuration file\n");
+            sb.append(ENV_VAR_ANDROID_BOOTCLASSPATH + " : android classpath\n");
+            sb.append(ENV_VAR_TRANSFORM_DEPENDENCIES_FILE + " : transform dependencies [optional]\n");
+            sb.append(ENV_VAR_TRANSFORM_CLASS + " : full qualified name for transform class\n");
+            throw new IllegalArgumentException(sb.toString());
+        }
+
+        //Loading transform dependencies in this class loader to ensure this application to work properly
+        final String[] postProcessDependencies = readDependenciesFileFromEnvVar(ENV_VAR_TRANSFORM_DEPENDENCIES_FILE);
+        if (postProcessDependencies.length > 0) {
+            new SystemClassLoader().loadJarFiles(postProcessDependencies);
+        }
+
+        //Passing the dependencies to the transform runner.
+        main(new TransformRunnerProvider() {
+            @Override
+            public TransformRunner provide() {
+
+                //Reading config file.
+                String configFilePath = System.getenv(ENV_VAR_CONFIG_FILE);
```

Comment:
Lets do java system properties instead of env vars. Its more standard practice and Retrolambda already works this way. They are also more explicit and environment is more implicit than we want